pkg/api/client/dynamic/supported.go

```diff
@@ -0,0 +1,50 @@
+package dynamic
+
+import "strings"
+
+const (
+	PrometheusRuleKind = "PrometheusRule"
+	ServiceMonitorKind = "ServiceMonitor"
+)
+
+// SupportedResource is an interface that returns information about Dynamic Resource that OLM supports.
+type SupportedResource interface {
+	IsNamespaced() bool
+}
+
+// supportedResource contains information about Dynamic Resource that OLM supports.
+type supportedResource struct {
+	name       string
+	namespaced bool
+}
+
+// IsNamespaced returns whether or not the supportedResource is a namespaced object.
+func (sr supportedResource) IsNamespaced() bool {
+	return sr.namespaced
+}
+
+var (
+	// supportedResources is a map whose keys are Kubernetes kinds and values are supportedResources.
+	supportedResources map[string]supportedResource
+)
+
+func init() {
+	supportedResources = make(map[string]supportedResource, 2)
+	// Supported Namespaced Resources
+	supportedResources[PrometheusRuleKind] = supportedResource{kindToResource(PrometheusRuleKind), true}
+	supportedResources[ServiceMonitorKind] = supportedResource{kindToResource(ServiceMonitorKind), true}
+}
+
+// TODO: Resolve resource based on GroupVersion, could be done with a partial group version https://github.com/kubernetes/apimachinery/blob/3253b0a30d67e7e362b8615e463156bac729c82f/pkg/api/meta/interfaces.go#L124
+func kindToResource(s string) string {
+	return strings.ToLower(s) + "s"
+}
+
+// IsSupported returns the a SupportedResource pointer if the kind provided is supported and nil otherwise
+func IsSupported(kind string) SupportedResource {
```

Comment:
is it better to return err if the kind is not supported?

Reply:
I do not see any value add in returning an error (or boolean).

By returning a pointer - we can convey if the object was present or not in the map.